Auth0/OAuth2Grant.swift

```diff
@@ -30,9 +31,34 @@ protocol OAuth2Grant {
 
 struct ImplicitGrant: OAuth2Grant {
 
-    let defaults: [String : String] = ["response_type": "token"]
+    let defaults: [String : String]
+    let responseType: [ResponseOptions]
+
+    init(responseType: [ResponseOptions] = [.token], nonce: String? = nil) {
+        self.responseType = responseType
+        if let nonce = nonce {
+            self.defaults = ["nonce" : nonce]
+        } else {
+            self.defaults = [:]
+        }
+    }
 
     func credentials(from values: [String : String], callback: @escaping (Result<Credentials>) -> ()) {
+        // id token reponse expectations and JWT validation, nonce validation
+        if responseType.contains(.id_token) {
```

Comment:
Move all logic of nonce validation to he validate method and then you can join guard and if into a guard.